### PR TITLE
Add more details to the goals config description

### LINKED ISSUE #1856: Add more docs to clarify the difference between goals/default.goals/hard.goals
For me, it's hard to understand the difference between these 3 configurations:

1. goals
2. default.goals
3. hard.goals

Also there are some hidden rules there: e.g. hard goals has to be in default goals list. 

I believe it would be very helpful to explicitly clarify more details on their differences, and how they are used by cruise-control.

### PR DESCRIPTION
**Pull request by CCisGG**
This PR resolves #1856.

Besides adding some description in the docs, this PR moves sanity check for self_healing goals from AnamolyDetectorManager  to KafkaCruiseControlConfig, as it seems to be cleaner.

### CODE REVIEW

**Review by hshi2022 — APPROVED** [review]

File: config/cruisecontrol.properties [hunk 1]
```diff
@@ -101,7 +101,7 @@ goals=com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal,com.linkedin
 # The list of supported intra-broker goals
 intra.broker.goals=com.linkedin.kafka.cruisecontrol.analyzer.goals.IntraBrokerDiskCapacityGoal,com.linkedin.kafka.cruisecontrol.analyzer.goals.IntraBrokerDiskUsageDistributionGoal
 
-# The list of supported hard goals -- consider using RackAwareDistributionGoal instead of RackAwareGoal in clusters with partitions whose replication factor > number of racks
+# The list of supported hard goals -- consider using RackAwareDistributionGoal instead of RackAwareGoal in clusters with partitions whose replication factor > number of racks. The value must be a subset of the "goals" and a superset of the "hard.goals" and "self.healing.goals".
```
Comment (efeg) [inline config/cruisecontrol.properties:104]: Nit: `...and a superset of the "hard.goals" and "self.healing.goals"` -> `...and a superset of the "self.healing.goals"`?

**Review by efeg — COMMENTED** [review]
Left some nits - o/w LGTM.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java [hunk 2]
```diff
@@ -225,7 +232,7 @@ private void sanityCheckGoalNames() {
    */
   void sanityCheckConcurrency() {
     int maxClusterMovementConcurrency = getInt(ExecutorConfig.MAX_NUM_CLUSTER_MOVEMENTS_CONFIG);
-    
+
```
[the remark this answers is not in the document]
Comment (CCisGG) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java:235]: Yeah it's not necessary, but these white spaces should be removed though?

File: config/cruisecontrol.properties (hunk 1 above)
Comment (CCisGG) [reply config/cruisecontrol.properties:104]: Hi Efe, do you mean it's not a superset of the hard.goals?

Comment (efeg) [reply config/cruisecontrol.properties:104]: This documentation is for `hard.goals` -- and it implies that the value of `hard.goals` must be a superset of the "hard.goals" (?).

Am I missing something?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (CCisGG) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java:235]: Yeah I think I'm using the style, and that's why this change is automatically made here. It's probably someone else make those white spaces and this change should fix it.

File: config/cruisecontrol.properties (hunk 1 above)
Comment (CCisGG) [reply config/cruisecontrol.properties:104]: My bad. Those comments should be for default goals. Will fix

**Review by efeg — APPROVED** [review]
Thanks for the updates. LGTM.

**Pull request merged**